domino/domino.py

```diff
@@ -264,21 +275,75 @@ def collaborators_add(self, usernameOrEmail, message=""):
         else:
             return disposition
 
-        
     # App functions
     def app_publish(self, unpublishRunningApps=True):
-        if unpublishRunningApps == True:
+        if unpublishRunningApps is True:
             self.app_unpublish()
         url = self._routes.app_publish()
         request = {"language": "App"}
         response = requests.post(url, auth=('', self._api_key), json=request)
         return response
-    
+
     def app_unpublish(self):
         apps = [r for r in self.runs_list()['data'] if r['notebookName'] == 'App' and r['isCompleted'] == False]
         for app in apps:
             self.run_stop(app['id'])
-    
+
+    # Environment functions
+    def environments_list(self):
+        self.requires_at_least("2.4.0")
+        url = self._routes.environments_list()
+        return self._get(url)
+
+    # Model Manager functions
+    def models_list(self):
+        self.requires_at_least("2.4.0")
+        url = self._routes.models_list()
+        return self._get(url)
+
+    def model_publish(self, file, function, environment_id, name,
+                      description="A great model", files_to_exclude=[]):
+        self.requires_at_least("2.4.0")
+
+        url = self._routes.model_publish()
+
+        request = {
+            "name": name,
+            "description": description,
+            "projectId": self._project_id,
+            "file": file,
+            "function": function,
+            "excludeFiles": files_to_exclude,
+            "environmentId": environment_id
+        }
+
+        response = requests.post(url, auth=('', self._api_key), json=request)
+        return response.json()
+
+    def model_versions_get(self, model_id):
+        self.requires_at_least("2.4.0")
+        url = self._routes.model_versions_get(model_id)
+        return self._get(url)
+
+    def model_version_publish(self, model_id, file, function, environment_id,
+                              name, description="A great model",
```

Comment:
Description default seems a bit gimicky and unnecessary. Feels like it would be annoying to set that as blank all the time.

Reply:
Yeah - #weekendCoding. Will set to blank